services/generic/meter.md

```diff
@@ -1,91 +1,170 @@
-`p_import_max`  | W       | Power Import max that day
-`p_export`      | W       | Power Export
-`p_export_react` | VAR    | Reactive Power Export
-`p_export_min`  | W       | Power Export minimum that day
-`p_export_max`  | W       | Power Export max that day
-`p_factor`      |         | Power Factor
-`freq`          | Hz      | Frequency
-`freq_min`      | Hz      | Frequency Min
-`freq_max`      | Hz      | Frequency Max
-`u1`            | V       | Voltage phase 1
-`u2`            | V       | Voltage phase 2
-`u3`            | V       | Voltage phase 3
-`i1`            | A       | Current phase 1
-`i2`            | A       | Current phase 2
-`i3`            | A       | Current phase 3
-`dc_p`          | W       | DC Power
-`dc_p_min`      | W       | DC Power
-`dc_p_max`      | W       | DC Power
-`dc_u`          | V       | DC Voltage
-`dc_u_min`      | V       | DC Min Voltage
-`dc_u_max`      | V       | DC Max Voltage
-`dc_i`          | A       | DC Current
-`dc_i_min`      | A       | DC Min Current
-`dc_i_max`      | A       | DC Max Current
\ No newline at end of file
+# Meter Services
+
+Meters report readings of imported/consumed or exported/produced values of their respective type.
+An electricity meter service can represent a stand-alone AMS meter, like a HAN sensor or a main electricity meter, or just a metering functionality of another device.
+
+## Service names
+
+| Service name    | Example units                                 | Description        |
+|-----------------|-----------------------------------------------|--------------------|
+| `meter_elec`    | W, kWh, V, A, kVAh, Hz, power_factor, pulse_c | Electricity meter. |
+| `meter_gas`     | cub_m, cub_f, pulse_c                         | Gas meter.         |
+| `meter_water`   | cub_m, cub_f, gallon, pulse_c                 | Water meter.       |
+| `meter_heating` | kWh                                           | Heating meter.     |
+| `meter_cooling` | kWh                                           | Cooling meter.     |
+
+## Interfaces
+
+| Type | Interface                   | Value type | Properties                                            | Storage          | Description                                                                                                                                                                            |
+|------|-----------------------------|------------|-------------------------------------------------------|------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| in   | cmd.meter.get_report        | string     |                                                       |                  | Value is an import/consumption `unit` defined in <br/>[`sup_units`](#service-properties) property. May not be supported by all meters. Empty value should request all supported units. |
```

Comment:
Empty value requests all supported units. 'Should' means in specs that sth is recommended byt not mandatory. Shall means its mandatory to be met.

Reply:
Good catch.